src/algebra/module/submodule_lattice.lean

```diff
@@ -159,4 +159,31 @@ show s ≤ Sup S, from le_Sup hs
 
 end submodule
 
+section nat_submodule
+
+/-- a submonoid is equivalent to a ℕ submodule -/
+def add_submonoid.to_nat_submodule : add_submonoid M ≃o submodule ℕ M :=
+{ to_fun := λ S,
+  { smul_mem' := λ r s hs, S.nsmul_mem hs _, ..S },
+  inv_fun := submodule.to_add_submonoid,
+  left_inv := λ ⟨S, _, _⟩, rfl,
+  right_inv := λ ⟨S, _, _, _⟩, rfl,
+  map_rel_iff' := λ a b, iff.rfl }
+
+@[simp]
+lemma add_submonoid.coe_to_nat_submodule (S : add_submonoid M) :
+  (S.to_nat_submodule : set M) = S := rfl
```

Comment:
I think this needs one more simp lemma:
```suggestion
@[simp]
lemma add_submonoid.to_nat_submodule_symm :
  ⇑(add_submonoid.to_nat_submodule.symm : _ ≃o add_submonoid M) = submodule.to_add_submonoid := rfl
  
@[simp]
lemma add_submonoid.coe_to_nat_submodule (S : add_submonoid M) :
  (S.to_nat_submodule : set M) = S := rfl
```
That may need line wrapping or extra type ascriptions